samigo/samigo-impl/src/java/org/sakaiproject/samigo/impl/SamigoAvailableNotificationServiceImpl.java

```diff
@@ -0,0 +1,369 @@
+                body.append("<br>");
+            }
+            if (extension.getTimeHours()==0 && extension.getTimeMinutes()==0) {
+                body.append(rl.getString("email.reminder.nolimit"));
+            } else {
+                body.append(rl.getFormattedMessage("email.reminder.limit", getTimeLimit(convertToSeconds(extension.getTimeHours(),extension.getTimeMinutes()))));
+            }
+        } else {    //normal
+            body.append(rl.getFormattedMessage("email.reminder.available", publishedAssessment.getTitle(), prettyDate(publishedAssessment.getStartDate().toInstant())));
+            body.append("<br>");
+            if(publishedAssessment.getDueDate() != null){
+                body.append(rl.getFormattedMessage("email.reminder.due", prettyDate(publishedAssessment.getDueDate().toInstant())));
+                body.append("<br>");
+            }
+            if (publishedAssessment.getTimeLimit() == 0) {
+                body.append(rl.getString("email.reminder.nolimit"));
+            } else {
+                body.append(rl.getFormattedMessage("email.reminder.limit", getTimeLimit(publishedAssessment.getTimeLimit())));
+            }
+        }
+        //todo: find how to get takebyurl link
+        String times = "an unlimited number of";
+        if (!publishedAssessment.getAssessmentAccessControl().getUnlimitedSubmissions()){
+            times = publishedAssessment.getAssessmentAccessControl().getSubmissionsAllowed().toString();
+        }
+        body.append(rl.getFormattedMessage("email.reminder.submit", times));
+        String score = "highest";
+        if (publishedAssessment.getEvaluationModel().getScoringType() == AVERAGE_SCORE){
+            score = "average";
+        } else if (publishedAssessment.getEvaluationModel().getScoringType() == LAST_SCORE){
+            score = "last";
+        } else if(publishedAssessment.getEvaluationModel().getScoringType() == ALL_SCORE){
+            score = "collective";
+        }
+        body.append(rl.getFormattedMessage("email.reminder.score", score));
+        body.append("<br>");
+        String feedback = " ";
+        if(publishedAssessment.getAssessmentFeedback().getFeedbackDelivery() == NO_FEEDBACK){ //3 appears to indicate No Feedback
+            feedback = " no ";
+        }
+        body.append(rl.getFormattedMessage("email.reminder.feedback", feedback));
+        body.append("<br>");
+        body.append(rl.getFormattedMessage("email.reminder.find", site.getTitle(), site.getUrl(), site.getUrl()));
+        Preferences userPrefs = preferencesService.getPreferences(member.getUserId());
+        ResourceProperties samigoOpenNoti = userPrefs.getProperties(NotificationService.PREFS_TYPE + SamigoConstants.NOTI_PREFS_TYPE_SAMIGO_OPEN);    // userPrefs.getProperties("noti:types:sakai:samigoOpen")
+        if(!StringUtils.equals((String) samigoOpenNoti.get("2"), "1")){    //don't send unless the user's preference allows it.
```

Comment:
Look to see if there are existing public static variable already in samigo that encapsulate these two values: "2", "1". If there aren't any already, create named variables for them so that this is easier to read and understand.